src/Processing.js

```diff
@@ -4066,10 +4066,36 @@
     p.noLoop = function() {
       doLoop = false;
       loopStarted = false;
-      clearInterval(looping);
+      
+      looping = false;
       curSketch.onPause();
     };
 
+requestAnimationFrame= (function(){
+      return window.requestAnimationFrame ||
+             window.webkitRequestAnimationFrame ||
+             window.mozRequestAnimationFrame    ||
+             function( callback ){
+                var _h= function(){callback(Date.now());};
+                window.setTimeout(_h, curMsPerFrame);
+             };
+     })();
+
+    var _now, _then=0, _doDraw = function(t) {
+      requestAnimationFrame(_doDraw);
+        if (!looping) { return; }
+        _now= t;
+        var _delta= _now - _then;
+        if (_delta > curMsPerFrame) {  // regulate the framerate
+            _then= _now - (_delta % curMsPerFrame);
```

Comment:
Pjs uses a 2 space indent, so if we can bump this back a little to match, that'll be worth doing.